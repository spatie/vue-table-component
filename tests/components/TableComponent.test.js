import Vue from 'vue/dist/vue.js';
import simulant from 'simulant';

describe('TableComponent', () => {
    beforeEach(() => {
        localStorage.clear();
    });

    it('can mount', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="[{ firstName: 'John' },{ firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can display nested properties', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="[{nested: { firstName: 'John' } }, { nested: { firstName: 'Paul' }}]">
                    <table-column show="nested.firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('accepts a function to format values', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="[{ firstName: 'John' },{ firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name" :formatter="formatter"></table-column>
                </table-component>
            </div>
        `;

        await createVm({
            methods: {
                formatter(value, properties) {
                    return `Formatted: <strong>${value}</strong>`;
                },
            },
        });

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('supports a scoped slot inside the table column', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="[{ firstName: 'John' },{ firstName: 'Paul' }]">
                    <table-column label="First name">
                        <template slot-scope="row">
                           {{ row.firstName }} slot
                        </template>
                    </table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('supports a named slot to display a tfoot section', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="[{ firstName: 'John' },{ firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name"></table-column>
                    <template slot="tfoot" slot-scope="{ rows }">
                        <tr>
                            <td>Name count:</td>
                            <td>{{ rows.length }}</td>
                        </tr>
                    </template>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('has an prop to disable the filter', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component :show-filter="false"
                    :data="[{ firstName: 'John' },{ id: 2, firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('has an prop to disable the caption', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component :show-caption="false"
                    :data="[{ firstName: 'John' },{ id: 2, firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('will use the property name as a column heading if label is not set', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component :show-filter="false"
                    :data="[{ firstName: 'John' },{ id: 2, firstName: 'Paul' }]">
                    <table-column show="firstName"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('won\'t use the property name as a column heading if label is an empty string', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component :show-filter="false"
                    :data="[{ firstName: 'John' },{ id: 2, firstName: 'Paul' }]">
                    <table-column show="firstName" label=""></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can display a custom message when filtering results in no results', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :extra-settings="{ labels: { filterResultEmpty: 'game over man, game over' } }"
                    :data="[{ firstName: 'John' },{ id: 2, firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        const table = await createVm();

        table.filter = 'this returns nothing';

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can display a custom placeholder in the filter field', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :extra-settings="{ labels: { filterPlaceholder: 'custom placeholder' } }"
                    :data="[{ firstName: 'John' },{ id: 2, firstName: 'Paul' }]">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        const table = await createVm();

        table.filter = 'this returns nothing';

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can accept a function to fetch the data', async () => {
        const serverResponse = () => {
            return {
                data: [{ firstName: 'John' }, { id: 2, firstName: 'Paul' }],
            };
        };

        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="${serverResponse}">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can render pagination when the server responds with pagination data', async () => {
        const serverResponse = () => {
            return {
                data: [{ firstName: 'John' }, { id: 2, firstName: 'Paul' }],

                pagination: {
                    totalPages: 4,
                    currentPage: 2,
                },
            };
        };

        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="${serverResponse}">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('clicking a link in the pagination will rerender the table', async () => {
        const serverResponse = ({ page }) => {
            return {
                data: [{ firstName: `John ${page}` }, { id: 2, firstName: `Paul ${page}` }],

                pagination: {
                    totalPages: 4,
                    currentPage: page,
                },
            };
        };

        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="${serverResponse}">
                    <table-column show="firstName" label="First name"></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();

        const thirdPageLink = document.getElementsByClassName('page-link')[2];

        await simulant.fire(thirdPageLink, 'click');

        await Vue.nextTick();
        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can add extra classes to the table, the cells and the headers', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component
                    :data="[{ firstName: 'John' },{ firstName: 'Paul' }]"
                    table-class="my-table"
                    thead-class="my-thead"
                    tbody-class="my-tbody"
                >
                    <table-column
                        show="firstName"
                        label="First name"
                        header-class="my-header"
                        cell-class="my-cell"
                    ></table-column>
                </table-component>
            </div>
        `;

        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can update columns', async () => {
        document.body.innerHTML = `
            <div id="app">
                <table-component :data="[{ firstName: 'John' },{ firstName: 'Paul' }]">
                    <table-column show="firstName" :label="label"></table-column>
                </table-component>
            </div>
        `;

        const vm = new Vue({
            el: '#app',
            data: {
                label: 'First name',
            },
        });

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();

        vm.label = 'Something else';

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm(options = {}) {
    const vm = new Vue({
        el: '#app',
        ...options,
    });

    await Vue.nextTick(() => {
    });

    const table = vm.$children[0];

    return table;
}
