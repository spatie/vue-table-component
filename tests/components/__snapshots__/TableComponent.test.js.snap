// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent accepts a function to format values 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              Formatted:
              <strong>
                John
              </strong>
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Formatted:
              <strong>
                Paul
              </strong>
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can accept a function to fetch the data 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can add extra classes to the table, the cells and the headers 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table my-table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head my-thead">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort my-header"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body my-tbody">
          <tr class="table-component__table__body__row">
            <td class="my-cell">
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td class="my-cell">
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can display a custom message when filtering results in no results 1`] = `

<div id="app">
  <div class="table-component"
       extra-settings="[object Object]"
  >
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <a class="table-component__filter__clear">
        ×
      </a>
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <div class="table-component__message">
      There are no matching rows
    </div>
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can display a custom placeholder in the filter field 1`] = `

<div id="app">
  <div class="table-component"
       extra-settings="[object Object]"
  >
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <a class="table-component__filter__clear">
        ×
      </a>
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <div class="table-component__message">
      There are no matching rows
    </div>
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can display nested properties 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can mount 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can render pagination when the server responds with pagination data 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <nav>
      <ul class="pagination justify-content-center">
        <li class="page-item">
          <a class="page-link">
            1
          </a>
        </li>
        <li class="page-item active">
          <a class="page-link">
            2
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            3
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            4
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>

`;

exports[`TableComponent can update columns 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent can update columns 2`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              Something else
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent clicking a link in the pagination will rerender the table 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John 1
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul 1
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <nav>
      <ul class="pagination justify-content-center">
        <li class="page-item active">
          <a class="page-link">
            1
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            2
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            3
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            4
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>

`;

exports[`TableComponent clicking a link in the pagination will rerender the table 2`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John 3
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul 3
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <nav>
      <ul class="pagination justify-content-center">
        <li class="page-item">
          <a class="page-link">
            1
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            2
          </a>
        </li>
        <li class="page-item active">
          <a class="page-link">
            3
          </a>
        </li>
        <li class="page-item">
          <a class="page-link">
            4
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>

`;

exports[`TableComponent has an prop to disable the caption 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <!---->
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent has an prop to disable the filter 1`] = `

<div id="app">
  <div class="table-component">
    <!---->
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent supports a named slot to display a tfoot section 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td>
              Name count:
            </td>
            <td>
              2
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent supports a scoped slot inside the table column 1`] = `

<div id="app">
  <div class="table-component">
    <div class="table-component__filter">
      <input type="text"
             placeholder="Filter table…"
             class="table-component__filter__field"
      >
      <!---->
    </div>
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              First name
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John slot
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul slot
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent will use the property name as a column heading if label is not set 1`] = `

<div id="app">
  <div class="table-component">
    <!---->
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
              firstName
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;

exports[`TableComponent won't use the property name as a column heading if label is an empty string 1`] = `

<div id="app">
  <div class="table-component">
    <!---->
    <div class="table-component__table-wrapper">
      <table class="table-component__table">
        <caption role="alert"
                 aria-live="polite"
                 class="table-component__table__caption"
        >
          Table not sorted
        </caption>
        <thead class="table-component__table__head">
          <tr>
            <th role="columnheader"
                aria-sort="none"
                class="table-component__th table-component__th--sort"
            >
            </th>
          </tr>
        </thead>
        <tbody class="table-component__table__body">
          <tr class="table-component__table__body__row">
            <td>
              John
            </td>
          </tr>
          <tr class="table-component__table__body__row">
            <td>
              Paul
            </td>
          </tr>
        </tbody>
        <tfoot>
        </tfoot>
      </table>
    </div>
    <!---->
    <div style="display: none;">
      <!---->
    </div>
    <!---->
  </div>
</div>

`;
